### DISCUSSION TITLE
xstate.assign event

### DISCUSSION THREAD

**Discussion by gkiely** [statelyai/xstate#3686] [Ideas]
A lot of the time when I assign a new value in an invoked actor, I would like to reflect some portion of that in the parent.

When using `sync: true`, xstate.update fires for every change to an invoked actor. It would be nice if there was an `xstate.assign` or `xstate.context` event and `xstate.state` to distinguish between a change to context and state.

Alternatively, something like `assignAndSendToParent` would also be helpful.

**Comment by davidkpiano** [comment]
Interesting idea; with observables as an analogy, you basically want to map/filter over the "emitted" values of the actor, and listen for those specific values, right?

One nice addition that is coming to v5 is `onSnapshot`, which is like `onDone` and `onError` but for emitted (snapshot) values from actors:

```js
// TENTATIVE v5 API
invoke: {
  src: someMachine,
  onSnapshot: {
    actions: (ctx, { data }) => {
      // `data` is the snapshot value
    }
  }
}
```

So that would be the first step to doing what you want; you can add a `guard` to help constrain but it looks like you want something more like a "selector".

Alright, this gives me some good food for thought.

**Reply by gkiely** [reply]
I think this would let me achieve what I’m trying to do.
